presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -85,6 +89,8 @@ public BigintGroupByHash(int hashChannel, boolean outputRawHash, int expectedSiz
 
         valuesByGroupId = new LongBigArray();
         valuesByGroupId.ensureCapacity(hashCapacity);
+
+        this.updateReservedMemory = requireNonNull(updateReservedMemory, "updateReservedMemory is null");
```

Comment:
why don't we reserve memory for the data structures that we allocate in this constructor (`hashCapacity`-sized stuff)?